src/python/pants/backend/python/util_rules/pex_requirements.py

```diff
@@ -47,6 +47,14 @@
 logger = logging.getLogger(__name__)
 
 
+@dataclass(frozen=True)
+class Resolve:
+    # A named resolve for a "user lockfile".
+    # Soon to be the only kind of lockfile, as this class will help
+    # get rid of the "tool lockfile" concept.
+    name: str
```

Comment:
After we get rid of tool lockfiles we may be able to pare the lockfile-related types (Resolve, ChosenPythonResolve, Lockfile, LoadedLockfileRequest, LoadedLockfile, EntireLockfile) down substantially.